### ISSUE TITLE
Custom branding: add validation to the plugin's settings

### ISSUE THREAD

**Issue by Desvelao** [wazuh/wazuh-kibana-app#4446]
## Description

We'll add validation to the settings of the plugin, so no invalid values can be set in the `Settings / Configuration` section.

## Objective
- Display errors when the input is invalid.
- Highlight the settings which contain errors.
- Disable configuration saving when there are errors.

## Tasks
- Frontend
    - [x] Add validation to each type of input component when its value changes.
    - [x] Highlight the errors of each input.
      - [x] Display the cause of error.
      - [x] Add warning icon to the setting with the errors.
    - [x] Disable the save button when there are errors due to invalid values.
    - [x] Add unit tests
    - [x] Add unit tests for `InputForm` of each setting.
- Backend
  - [x] Validate route and body parameters of the following endpoints:
    - [x] `PUT /utils/configuration`
    - [x] `PUT /utils/configuration/files/{key}`
  - [x] Tests
    
**Depends on**: #4443

**Comment by Desvelao** [comment]
# WIP

## Enhance the managing of the input state of the form

I refactored the inputs are managed and the global state of the form in `SettingsConfiguration` with the next changes:
- Create hooks to manage the state of the input and global form state to support the error reporting.

  > These hooks are focused to report if the field was changed compared to an initial value.
  
- Add to the input a message when its value is invalid.
- Add highlighting what setting has an invalid value.
- Add the count of errors in the bar to save the configuration.
- Disable the save button if there is some error in the inputs due to an invalid value.

**Screenshots**
- Validation error, highlighting the setting with invalid value, display the count of inputs with invalid value and disable the save button
![image](https://user-images.githubusercontent.com/34042064/188813569-958c5d6d-d66c-4840-b805-0df33b283d64.png)

- Hover the icon displayed in the setting when its value is invalid
![image](https://user-images.githubusercontent.com/34042064/188813794-c49c7e11-26e5-4649-927b-9dbfb54943e6.png)

**Comment by Desvelao** [comment]
# WIP

- Added validation to each setting

**Comment by Desvelao** [comment]
# Changes

- Fix some settings validation errors
- Update the development branch with the last changes of the base branch

**Comment by Desvelao** [comment]
# Changes

- Add tests to validate the updating of plugin settings using the API endpoint `PUT /utils/configuration`. (WIP)

**Comment by Desvelao** [comment]
In the validation, we should consider the values used by cloud environments, to be sure these pass the setting validations.

Some plugin configurations to keep in mind:
- https://wazuh-team.slack.com/archives/GPAMUL65D/p1665130509029029?thread_ts=1665130129.243349&cid=GPAMUL65D

**Comment by asteriscos** [comment]
I created a new PR related to this issue to add de validation description to multi-line inputs description.

![image](https://user-images.githubusercontent.com/9343732/199015232-baf5ac4e-4eb8-44a0-a967-65f2cf3828ad.png)
